src/app/store/slices/storage/index.ts

```diff
@@ -161,7 +182,7 @@ export const storageSlice = createSlice({
       action: PayloadAction<{ updateRecents?: boolean; folderIds?: number[]; items: DriveItemData | DriveItemData[] }>,
     ) {
       const itemsToPush = !Array.isArray(action.payload.items) ? [action.payload.items] : action.payload.items;
-      const folderIds = action.payload.folderIds || Object.keys(state.levels).map((folderId) => parseInt(folderId));
+      const folderIds = !Array.isArray(action.payload.folderIds) ? [action.payload.folderIds] : (action.payload.folderIds || Object.keys(state.levels).map((folderId) => parseInt(folderId)));
```

Comment:
I would argue this is very confusing to read perhaps create an intermediate variable with a descriptive name or invert the ternary to not use `!`